### PR TITLE
feat: add comment line ability

### PR DESCRIPTION
**Pull request by grianbrcom**
Signed-off-by: Anton Grigorev <anton.grigorev@broadcom.com>

### CODE REVIEW

File: clients/cobol-lsp-vscode-extension/src/commands/CommentCommand.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    private handleSelection(selection: vscode.Selection) {
+        const selectedLines = this.getSelectedLines(selection)
+        const textLines = selectedLines.map(it => it.text);
+        const anyNonComment = textLines
+            .map(getLineCommentStatus)
+            .some(it => it === LineCommentStatus.NON_COMMENT);
+        const action = anyNonComment ? commentLine : uncommentLine;
+        const processedLines = textLines.map(action);
+        const lineSeparator = this.textEditor.document.eol === vscode.EndOfLine.LF ? '\n' : '\r\n';
```
Comment (temanbrcom) [inline clients/cobol-lsp-vscode-extension/src/commands/CommentCommand.ts:47]: So, it is a configuration of VSC, not of the file itself? Is it expected?

**Review by temanbrcom — COMMENTED** [review]
Did you test it with CBL/PROCESS directives? They may start before the indicator area.

File: clients/cobol-lsp-vscode-extension/package.json [hunk 2]
```diff
@@ -54,6 +54,10 @@
             {
                 "command": "cobol-lsp.cpy-manager.goto-settings",
                 "title": "Open Copybook location settings"
+            },
+            {
+                "command": "cobol-lsp.action.commentLine",
+                "title": "Comment line"
```
Comment (VitGottwald) [inline clients/cobol-lsp-vscode-extension/package.json:60]: Could we rename the title to `Toggle COBOL Line Comment`? That would make it more in line with the default `Toggle Line Comment`. We are also going to implement commenting for HLASM and I would call it `Toggle HLASM Line Comment` there to have some consistency if a user searches the command pallet.

**Comment by VitGottwald** [comment]
Looks good! There is also a command to `Add Line Comment` and `Remove Line Comment`. Would it be hard to add those (`Add COBOL Line Comment` and `Remove COBOL Line Comment`)?

File: clients/cobol-lsp-vscode-extension/src/commands/CommentCommand.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+import * as vscode from "vscode";
+
+export function commentCommand() {
```
[the remark this answers is not in the document]
Comment (grianbrcom) [reply clients/cobol-lsp-vscode-extension/src/commands/CommentCommand.ts]: Usually, we didn't do it for TS. But if you insist...

File: clients/cobol-lsp-vscode-extension/src/commands/CommentCommand.ts (hunk 1 above)
Comment (grianbrcom) [reply clients/cobol-lsp-vscode-extension/src/commands/CommentCommand.ts:54]: That comes from VSC. Have no idea how it evaluates it. You can see it in the bottom right part:
image
And even change manually.

File: clients/cobol-lsp-vscode-extension/package.json (hunk 2 above)
Comment (grianbrcom) [reply clients/cobol-lsp-vscode-extension/package.json]: Sure. Done.

**Comment by grianbrcom** [comment]
@VitGottwald, I've added two commands: `Comment COBOL Line` and `Uncomment COBOL Line`. In case the user wants to do particular action instead of toggling. For example when you toggle line comment for:
```
      *    DISPLAY 'From P1:' BAR of FOOBAR.
```
it becomes uncommented:
```
           DISPLAY 'From P1:' BAR of FOOBAR.
```
But if you run `Comment COBOL Line` instead, it will be:
```
      **    DISPLAY 'From P1:' BAR of FOOBAR.
```
And now I see that you mention Add/Remove line comment. Did you mean something different?

File: clients/cobol-lsp-vscode-extension/src/commands/CommentCommand.ts (hunk 1 above)
Comment (temanbrcom) [reply clients/cobol-lsp-vscode-extension/src/commands/CommentCommand.ts:54]: I just wasn't sure if we have any convention about this. I don't really care.

File: clients/cobol-lsp-vscode-extension/src/commands/CommentCommand.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (temanbrcom) [reply clients/cobol-lsp-vscode-extension/src/commands/CommentCommand.ts]: No, I don't really care. I just wasn't sure if we have any agreement about that.

File: clients/cobol-lsp-vscode-extension/src/commands/CommentCommand.ts (hunk 1 above)
Comment (temanbrcom) [reply clients/cobol-lsp-vscode-extension/src/commands/CommentCommand.ts:54]: Ok

**Comment by VitGottwald** [comment]
This is exactly what I meant. Only the names of the commands are different. What you call 

- `Comment COBOL Line` should be called `Add Cobol Line Comment` 
- `Uncomment COBOL Line` should be called `Remove COBOL Line Comment`

to be in line with the way VSCode hames these as `Add Line Comment` / `Remove Line Comment`.

**Comment by grianbrcom** [comment]
Got it. I'll change the names.

**Review by temanbrcom — APPROVED** [review]
💡

**Pull request merged**